### PR TITLE
Refactor Measurement

### PR DESCRIPTION
**Pull request by beomki-yeo**
Changes:
- Removed the `measurement_link` and add `detray::barcode` variable to measurement EDM
- Define `track_candidate` as `measurement`

### CODE REVIEW

File: core/include/traccc/finding/finding_algorithm.ipp [hunk 1]
```diff
@@ -152,7 +152,7 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
                                                    in_param.covariance());
                 const auto& meas = measurements_on_surface[item_id];
 
-                track_state<transform3_type> trk_state({module_id, meas});
+                track_state<transform3_type> trk_state({meas});
```
Comment (niermann999) [inline core/include/traccc/finding/finding_algorithm.ipp:155]: Or does this take something else that is constructible from a measurement?
```suggestion
                track_state<transform3_type> trk_state(meas);
```

File: device/common/include/traccc/finding/device/impl/find_tracks.ipp [hunk 2]
```diff
@@ -99,8 +99,7 @@ TRACCC_DEVICE inline void find_tracks(
 
         bound_track_parameters in_par = in_params.at(in_param_id);
         const auto meas = measurements_on_surface.at(i + stride);
-        track_state<typename detector_t::transform3> trk_state(
-            {module_id, meas});
+        track_state<typename detector_t::transform3> trk_state({meas});
```
Comment (niermann999) [inline device/common/include/traccc/finding/device/impl/find_tracks.ipp:102]: Same here
```suggestion
        track_state<typename detector_t::transform3> trk_state(meas);
```

File: io/include/traccc/io/event_map2.hpp [hunk 3]
```diff
@@ -52,13 +39,8 @@ struct event_map2 {
...
 
-            for (const auto& meas_link : measurements) {
-
-                track_candidate cand = {
-                    detray::geometry::barcode{meas_link.surface_link},
-                    meas_link.meas};
-
-                candidates.push_back(cand);
+            for (const auto& meas : measurements) {
+                candidates.push_back(meas);
```
Comment (niermann999) [inline io/include/traccc/io/event_map2.hpp:43]: Maybe unrelated to this PR, but should this get a call to ```reserve```?

**Review by niermann999 — CHANGES_REQUESTED** [review]
Looks good in general

File: device/common/include/traccc/finding/device/impl/find_tracks.ipp (hunk 2 above)
Comment (beomki-yeo) [reply device/common/include/traccc/finding/device/impl/find_tracks.ipp]: Fixed

File: io/include/traccc/io/event_map2.hpp (hunk 3 above)
Comment (beomki-yeo) [reply io/include/traccc/io/event_map2.hpp:44]: You are right. Fixed.

**Review by niermann999 — APPROVED** [review]

**Pull request merged**